packages/generator/templates/app/app/auth/pages/forgot-password.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from "react"
+import { BlitzPage, useRouterQuery, Link } from "blitz"
+import Layout from "app/layouts/Layout"
+import { LabeledTextField } from "app/components/LabeledTextField"
+import { Form, FORM_ERROR } from "app/components/Form"
+import {
+  ForgotPasswordInput,
+  ForgotPasswordInputType,
+  ResetPasswordInput,
+  ResetPasswordInputType,
+} from "app/auth/validations"
+import forgotPassword from "app/auth/mutations/forgotPassword"
+import resetPassword from "app/auth/mutations/resetPassword"
+
+/*
+ * By default this page renders the ForgotPasswordForm form which gets the user
+ * email and sends password reset instructions.
+ *
+ * When they click the link in that email, they will be sent back to this same
+ * page but with the token set as a query paramater like `?token=XXXXX`.
+ *
+ * The ResetPasswordForm is rendered if the `?token=` paramater is present in the URL
+ */
+
+const ForgotPasswordForm = () => {
+  const [success, setSuccess] = useState(false)
+  return (
+    <div>
+      <h1>Forgot your password?</h1>
+
+      {success ? (
+        <div>
+          <h2>Request Submitted</h2>
+          <p>
+            If your email is in our system, you will receive instructions to reset your password
+            shortly.
+          </p>
+        </div>
+      ) : (
+        <Form<ForgotPasswordInputType>
+          submitText="Send Reset Password Instructions"
+          schema={ForgotPasswordInput}
+          initialValues={{ email: "" }}
+          onSubmit={async (values) => {
+            try {
+              await forgotPassword(values)
+              setSuccess(true)
+            } catch (error) {
+              return {
+                [FORM_ERROR]: "Sorry, we had an unexpected error. Please try again.",
+              }
+            }
+          }}
+        >
+          <LabeledTextField name="email" label="Email" placeholder="Email" />
+        </Form>
+      )}
+    </div>
+  )
+}
+
+const ResetPasswordForm = () => {
+  const query = useRouterQuery()
+  const [success, setSuccess] = useState(false)
+  return (
+    <div>
+      <h1>Set a New Password</h1>
+
+      {success ? (
+        <div>
+          <h2>Password Reset Successfully</h2>
+          <p>
+            You may now <Link href="/login">log in with your new password</Link>
+          </p>
+        </div>
+      ) : (
+        <Form<ResetPasswordInputType>
+          submitText="Reset Password"
+          schema={ResetPasswordInput}
+          initialValues={{ password: "", passwordConfirmation: "", token: query.token as string }}
+          onSubmit={async (values) => {
+            try {
+              await resetPassword(values)
+              setSuccess(true)
```

Comment:
suggestion: why not just create a new session for the user at the end of resetPassword mutation and then redirect  to whatever the default logged in page is.  Saves the user from having to click login link and then login manually.  I know a lot of apps use the flow you have here but I can't really think of a benefit of making the user go through these extra steps.

Reply:
Because of this [from here](https://cheatsheetseries.owasp.org/cheatsheets/Forgot_Password_Cheat_Sheet.html).
> Once they have set their new password, the user should then login through the usual mechanism. Don't automatically log the user in, as this introduces additional complexity to the authentication and session handling code, and increases the likelihood of introducing vulnerabilities.

But now that I'm thinking about it more, I think we can do it automatically since our code for creating sessions is so simple (1 line)

Reply:
Ah ok hadn't read that.  It does seem fairly low risk and offers imo a better end-user experience 🤷‍♂️

Reply:
I'm changing it to automatically log you in